File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/HighestLayPriceFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.Side;
import org.springframework.stereotype.Component;

@Component
public class HighestLayPriceFunction extends AbstractPriceReduceFunction {

    public HighestLayPriceFunction() {
        super(Side.LAY, Math::max);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/HighestLayPriceFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.Side
import org.springframework.stereotype.Component
import kotlin.math.max

@Component
object HighestLayPriceFunction : ProgressFunction by PriceReduceFunction(Side.LAY, { a, b -> max(a, b) })
